internal/tiltfile/docker_compose.go

```diff
@@ -37,33 +40,72 @@ type dcResourceSet struct {
 func (dc dcResourceSet) Empty() bool { return reflect.DeepEqual(dc, dcResourceSet{}) }
 
 func (s *tiltfileState) dockerCompose(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	configPaths := value.NewLocalPathListUnpacker(thread)
+	var configPaths starlark.Value
 
 	err := s.unpackArgs(fn.Name(), args, kwargs, "configPaths", &configPaths)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, v := range configPaths.Value {
-		err = io.RecordReadPath(thread, io.WatchFileOnly, v)
-		if err != nil {
-			return nil, err
-		}
+	paths := starlarkValueOrSequenceToSlice(configPaths)
+
+	if len(paths) == 0 {
+		return nil, fmt.Errorf("Nothing to compose")
 	}
 
 	dc := s.dc
+	project := model.DockerComposeProject{ConfigPaths: dc.configPaths, ProjectPath: dc.Project.ProjectPath}
+
 	currentTiltfilePath := starkit.CurrentExecPath(thread)
 	if dc.tiltfilePath != "" && dc.tiltfilePath != currentTiltfilePath {
 		return starlark.None, fmt.Errorf("Cannot load docker-compose files from two different Tiltfiles.\n"+
 			"docker-compose must have a single working directory:\n"+
 			"(%s, %s)", dc.tiltfilePath, currentTiltfilePath)
 	}
 
-	// To make sure all the docker-compose files are compatible together,
-	// parse them all together.
-	allConfigPaths := append([]string{}, dc.configPaths...)
-	allConfigPaths = append(allConfigPaths, configPaths.Value...)
-	project := model.DockerComposeProject{ConfigPaths: allConfigPaths}
+	for _, val := range paths {
+		switch v := val.(type) {
+		case nil:
+			continue
+		case io.Blob:
+			yaml := v.String()
+			message := "unable to store yaml blob"
+			tmpfile, err := os.CreateTemp("", fmt.Sprintf("%s-docker-compose-*.yml", filepath.Base(filepath.Dir(currentTiltfilePath))))
```

Comment:
One minor issue is tempfile proliferation -- we never clean them up. Is there an appropriate place that could happen, or do we just live with it?

Reply:
What springs to mind:
 * Could (lazily) create a tempdir for the current Tilt process first time needed (stored on `tiltfileState`) and then in the loader set up a long-running goroutine that just blocks on `ctx.Done()` and then deletes the whole dir
 * Name these files deterministically (e.g. hash the string) so that we don't keep creating new files every Tiltfile invocation, only when the blob changes

Reply:
Yeah, I had a previous version where I was hashing the string, that would definitely cut down on the number of files at least.